src/main/java/seedu/address/model/assessment/AssessmentResultsList.java

```diff
@@ -45,6 +47,39 @@ public boolean contains(AssessmentResults toCheck) {
         return assessmentResultsList.stream().anyMatch(toCheck::isSameAssessmentResults);
     }
 
+    /**
+     * Returns true if assessmentResultsList contains an AssessmentResult with the given name.
+     */
+    public boolean hasAssessmentResultsByName(AssessmentName assessmentName) {
+        requireNonNull(assessmentName);
+        return assessmentResultsList.stream().anyMatch(x -> x.hasAssessmentName(assessmentName));
+    }
+
+    /**
+     * Returns true if assessmentResultsList contains the result of a student with {@Code studentId} for
+     * the assessment with {@Code assessmentName}.
+     */
+    public boolean hasStudentResult(AssessmentName assessmentName, NusNetId studentId) {
```

Comment:
Can I check why did you choose to use NusNetId to implement hadStudentResult instead of using name? Many of our method uses name to reference a Person or Student. I don't see anything wrong with using NusNetId though!

Reply:
haha at the moment StudentResult stores by id instead
i was considering the change too, but we will need to change the StudentResult implementation